index.js

```diff
@@ -103,6 +105,22 @@ function fastifyWebsocket (fastify, opts, next) {
         throw new Error('invalid wsHandler function')
       }
     }
+    
+    // this is to add req.ws
+    function reqWsAdder(request, reply) {
+      if (request.raw[kWs]) {
+        request.ws = true;
+      } else {
+        request.ws = false;
+      }
+    }
+    if (Array.isArray(routeOptions.preHandler)) {
+      routeOptions.preHandler = [reqWsAdder, ...routeOptions.preHandler];
+    } else if (typeof routeOptions.preHandler === "function") {
+      routeOptions.preHandler = [reqWsAdder, routeOptions.preHandler];
+    } else {
+      routeOptions.preHandler = reqWsAdder;
```

Comment:
Instead of adding this to the route, what about a server level hook that sets this property above this route handler?You can add hooks right onto the server and they will run for all routes registered on that server instance, see https://www.fastify.io/docs/latest/Hooks/#onrequest. 

Also, would be great to get some tests!

Reply:
I don't understand what you're suggesting.

Reply:
Oh, I see. Thank you for the idea!